### PR TITLE
Warn about widget parameters that could not be applied by Param

### PR DESCRIPTION
**Pull request by philippjfr**
Add warning for cases like https://github.com/holoviz/panel/issues/4933

### CODE REVIEW

File: panel/param.py [hunk 1]
```diff
@@ -467,6 +467,13 @@ def widget(self, p_name):
             kw['description'] = textwrap.dedent(p_obj.doc).strip()
 
         # Update kwargs
+        ignored_kws = [repr(k) for k in kw_widget if k not in widget_class.param]
+        if ignored_kws:
+            self.param.warning(
+                f'Param pane was given unknown keyword argument(s) for {p_name!r} '
+                f'parameter with a widget of type {widget_class!r}. The following '
```
Comment (Hoxbro) [inline panel/param.py:474]:
```suggestion
                f'parameter with a widget of type {widget_class.__name___!r}. The following '
```

**Review by Hoxbro — COMMENTED** [review]
Is there a way to let `ignored_kws` be passed to the widget at instantiation and then let the widget handle it? Like passing them in here: 
https://github.com/holoviz/panel/blob/3b588a31d1af515ba8271a76ead7e9ad4b078dc2/panel/param.py#LL487C1-L487C44

**Comment by philippjfr** [comment]
In future Param will error if you give it unexpected kwargs so no we can't just pass it in but for situations like this it may indeed be nice to have a non-strict mode.

**Comment by Hoxbro** [comment]
I don't see a problem in erroring if the input is wrong. Passing in `configuration` into Tabulator would still work in future versions of Param.

**Comment by philippjfr** [comment]
Okay, I agree. Can't make that change in a micro release though.

**Pull request merged**